File: go-leetcode/linked-list/206.反转链表.go
package leetcode

func reverseList(head *ListNode) *ListNode {
	// 0 | 1 个节点
	if head == nil || head.Next == nil {
		return head
	}

	var pre *ListNode = nil
	curr := head

	for curr != nil {
		temp := curr.Next
		curr.Next = pre
		pre = curr
		curr = temp
	}

	return pre
}

File: python-leetcode/linked-list/206.反转链表.py
from typing import Optional

from type import ListNode


class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # 0 | 1 个节点
        if not head or not head.next:
            return head

        pre = None
        curr = head

        while curr:
            temp = curr.next
            curr.next = pre
            pre = curr
            curr = temp
        return pre
